### PR TITLE
[fix](array) fix array_except/union for left const return only one row result

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
backport : https://github.com/apache/doris/pull/36776

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/functions/array/function_array_set.h [hunk 1]
```diff
@@ -181,7 +181,11 @@ struct ArraySetImpl {
         constexpr auto execute_left_column_first = Impl::Action::execute_left_column_first;
         size_t current = 0;
         Impl impl;
-        for (size_t row = 0; row < left_data.offsets_ptr->size(); ++row) {
+	size_t row_size = left_data.offsets_ptr->size();
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_array_set.h:184]: fix clang format and add comment

Comment (amorynan) [reply be/src/vec/functions/array/function_array_set.h]: done!

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run external

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run external

**Pull request merged**